app/src/main/java/com/sma2/sma2/ExerciseFragments/ExerciseFragment.java

```diff
@@ -0,0 +1,44 @@
+package com.sma2.sma2.ExerciseFragments;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+
+
+public abstract class ExerciseFragment extends Fragment {
+    protected ExerciseFragment.OnFragmentInteractionListener mListener;
+    protected boolean recording = false;
+    protected String filePath;
+    protected String exercise;
+
+    public ExerciseFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof ExerciseFragment.OnFragmentInteractionListener) {
+            mListener = (ExerciseFragment.OnFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+    }
+
+    public interface OnFragmentInteractionListener {
```

Comment:
Maybe we can have a better name for this.

Reply:
Das stimmt. Wie wäre es mit OnExerciseCompletedListener? Der wird von mir ja nur gefeuert wenn die Exercise beendet ist.

Reply:
Jap Klingt gut